src/dataflow-types/src/client/controller.rs

```diff
@@ -41,6 +44,22 @@ impl<C: Client> Client for Controller<C> {
 impl<C: Client> Controller<C> {
     pub async fn send(&mut self, cmd: Command) {
         match &cmd {
+            Command::Compute(ComputeCommand::CreateInstance(logging), instance) => {
+                self.compute_since_uppers
+                    .insert(*instance, Default::default());
+
+                if let Some(logging_config) = logging {
+                    for id in logging_config.log_identifiers() {
+                        self.compute_since_uppers
+                            .get_mut(instance)
+                            .expect("Reference to absent instance")
```

Comment:
If this wasn't merged, I'd have asked to move the `get_mut` out of the loop, here and elsewhere. This can be a costly call.

Reply:
I can follow up with that. It was a minor nuisance because it prevents a `self.function` call up above, which made me unpick an abstraction. But 100% this can be outside all the loops.